### PR TITLE
feat: added roles to emojis

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR adds roles to emojis, allowing bots to restrict emojis to certain roles. You can view discord's documentation about this at [this page](https://discord.com/developers/docs/resources/emoji#emoji-object).

This also changes how we store the user, along with how we send information to discord. We now store the actual user object rather than a user_id (as discord provides the user object, this helps people who may not have cache on so they don't have to make another API request to get the user).

The information above is a breaking change.

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

**Comment by Jaskowicz1** [comment]
image

Here's an image of it working! In this image, I restricted an emoji to my role only, meaning Brain couldn't use the emoji "memelol" until I set the roles variable to a blank array.

**Review by braindigitalis — APPROVED** [review]

File: include/dpp/emoji.h [hunk 1]
```diff
@@ -71,22 +71,11 @@ class DPP_EXPORT emoji : public managed, public json_interface<emoji> {
...
-	snowflake user_id{0};
-	/**
-	 * @brief Flags for the emoji from dpp::emoji_flags
-	 */
-	uint8_t flags{0};
-	/**
-	 * @brief Image data for the emoji if uploading
-	 */
-	std::string image_data{};
+	std::string 			name{};		//!< emoji name
```
Comment (Mishura4) [inline include/dpp/emoji.h:74]: Why change the documentation formatting here?

**Review by Mishura4 — CHANGES_REQUESTED** [review]

File: include/dpp/emoji.h (hunk 1 above)
Comment (braindigitalis) [reply include/dpp/emoji.h:74]: yeah probably best not to, because we're getting rid of all the `//!<` formatting junk

Comment (Jaskowicz1) [reply include/dpp/emoji.h:74]: this was already agreed upon in discord that we would ignore this and sort it in a later PR.

Comment (Mishura4) [reply include/dpp/emoji.h]: But why does your PR change the style?
This is *your* PR introducing the style we want to stop using, this sounds like the perfect time to revert this change

**Review by Mishura4 — APPROVED** [review]

**Pull request merged**